command/bitmarkd/commands.go

```diff
@@ -131,52 +147,73 @@ func processSetupCommand(arguments []string) bool {
+		fmt.Printf("                                        			and the certificate in: %q\n", "DIR/"+rpcCertificateKeyFilename)
 		fmt.Printf("\n")
 
-		fmt.Printf("  gen-proof-identity [DIR]   (proof)  - create private key in: %q\n", "DIR/"+proofPrivateKeyFilename)
-		fmt.Printf("                                        the public key in:     %q\n", "DIR/"+proofPublicKeyFilename)
-		fmt.Printf("                                        and signing keys in:  %q and: %q\n", "DIR/"+proofLiveSigningKeyFilename, "DIR/"+proofTestSigningKeyFilename)
+		fmt.Printf("  gen-proof-identity d=[DIR] v=[VER]   	(proof)  		- create private key in: %q\n", "DIR/"+proofPrivateKeyFilename)
+		fmt.Printf("                                        			the public key in:     %q\n", "DIR/"+proofPublicKeyFilename)
+		fmt.Printf("                                        			and signing keys in:  %q and: %q\n", "DIR/"+proofLiveSigningKeyFilename, "DIR/"+proofTestSigningKeyFilename)
 		fmt.Printf("\n")
 
-		fmt.Printf("  dns-txt                    (txt)    - display the data to put in a dbs TXT record\n")
+		fmt.Printf("  dns-txt                    		(txt)    		- display the data to put in a dbs TXT record\n")
 		fmt.Printf("\n")
 
-		fmt.Printf("  start                      (run)    - just run the program, same as no arguments\n")
-		fmt.Printf("                                        for convienience when passing script arguments\n")
+		fmt.Printf("  start                      		(run)    		- just run the program, same as no arguments\n")
+		fmt.Printf("                                        			for convienience when passing script arguments\n")
 		fmt.Printf("\n")
 
-		fmt.Printf("  block S [E [FILE]]         (b)      - dump block(s) as a JSON structures to stdout/file\n")
+		fmt.Printf("  block S [E [FILE]]         		(b)      		- dump block(s) as a JSON structures to stdout/file\n")
 		fmt.Printf("\n")
 
-		fmt.Printf("  save-blocks FILE           (save)   - dump all blocks to a file\n")
+		fmt.Printf("  save-blocks FILE           		(save)   		- dump all blocks to a file\n")
 		fmt.Printf("\n")
 
-		fmt.Printf("  load-blocks FILE           (load)   - restore all blocks from a file\n")
-		fmt.Printf("                                        only runs if database is deleted first\n")
+		fmt.Printf("  load-blocks FILE           		(load)   		- restore all blocks from a file\n")
+		fmt.Printf("                                        			only runs if database is deleted first\n")
 		fmt.Printf("\n")
 
-		fmt.Printf("  delete-down NUMBER         (dd)     - delete blocks in descending order\n")
+		fmt.Printf("  delete-down NUMBER         		(dd)     		- delete blocks in descending order\n")
 		fmt.Printf("\n")
 
 		exitwithstatus.Exit(1)
 	}
 
+done:
 	// indicate processing complete and prefor normal exit from main
 	return true
 }
 
+func parseArgs(arguments []string) (map[string]string, error) {
+	arg := make(map[string]string)
+
+	for _, a := range arguments {
+		i := strings.Index(a, "=")
```

Comment:
I dont think it's different. Both of them are fine to me. I use index to detect the character is contained or not, instead of `strings.Contains()`.

Reply:
Hi, sorry to jump in this review, however strings.Contains runtime implementation will be faster than "for and range", the runtime implementation covers different cases to deal with sub-strings, it also uses IndexByte instead of IndexString, also it is more readable and you will end up with less code.

Reply:
@araujobsd I think @jamieabc mentioned about this line `i := strings.Index(a, "=")`, not for a loop because the loop is required for iterate all value from `arguments`. 
Anyway we always need comments from you.

Reply:
OK, in that case doesn't matter :)

Reply:
Anyway i need to remove this changes because of Chris's comment. Thanks @jamieabc  & @araujobsd